addons/web/static/src/core/overlay/overlay_container.xml

```diff
@@ -2,8 +2,8 @@
 <templates xml:space="preserve">
 
     <t t-name="web.OverlayContainer">
-        <div class="o-overlay-container">
-            <t t-foreach="sortedOverlays" t-as="overlay" t-key="overlay.id">
+        <div class="o-overlay-container" t-ref="root">
+            <t t-foreach="sortedOverlays" t-as="overlay" t-key="overlay.id" t-if="isVisible(overlay)">
```

Comment:
I would put `t-if` and `t-foreach` on different node, because it's ambiguous when these 2 directives are on same node: should it be read `if ... then foreach ...` or `foreach ... if ... then ...`?
Even if at a glance the later is the once working, I think this is an arbitrary choice and it can mistakenly understood by just looking at template at a glance.